src/lib/optimizer/strategy/projection_combination_rule.cpp

```diff
@@ -0,0 +1,106 @@
+#include "projection_combination_rule.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/projection_node.hpp"
+
+namespace opossum {
+
+std::string ProjectionCombinationRule::name() const { return "Projection Combination Rule"; }
+
+bool ProjectionCombinationRule::apply_to(const std::shared_ptr<AbstractLQPNode>& node) {
+  if (node->type() != LQPNodeType::Projection) {
+    return _apply_to_children(node);
+  }
+
+  std::vector<std::shared_ptr<ProjectionNode>> projection_nodes;
+
+  // Gather subsequent ProjectionNodes
+  auto current_node = node;
+  while (current_node->type() == LQPNodeType::Projection) {
+    projection_nodes.emplace_back(std::static_pointer_cast<ProjectionNode>(current_node));
+
+    // If current node has multiple or no children, end the combination here
+    if (current_node->right_child() || !current_node->left_child()) {
+      break;
+    }
+
+    current_node = current_node->left_child();
+
+    if (current_node->parents().size() > 1) {
+      break;
+    }
+  }
+
+  if (projection_nodes.size() > 1) {
+    const auto combined_projection_node = _combine_projections(projection_nodes);
+    _apply_to_children(combined_projection_node);
+    return true;
+  }
+
+  return _apply_to_children(node);
+}
+
+std::shared_ptr<ProjectionNode> ProjectionCombinationRule::_combine_projections(
+    std::vector<std::shared_ptr<ProjectionNode>>& projections) const {
```

Comment:
I'd call a single LQPExpression in a ProjectionNode a "projection" and would rather name this variable "projection_nodes"

Reply:
Sounds good